src/matrixClass.ts

```diff
@@ -0,0 +1,317 @@
+            //vert ship
+            console.log("vertical ")
+            return this.verticalShipSunk(lastShot);
+        }
+        else{
+            console.log("ship size 1")
+            //ship of size one
+            if(this.getRowPos(up.Row) <9 && !this.alreadyMissAt(up))this.board[this.getRowPos(up.Row)][up.Column-1] = 3;
+            else if (right.Column>0 && !this.alreadyMissAt(right))this.board[this.getRowPos(right.Row)][right.Column-1] = 3;
+            else if (this.getRowPos(down.Row)>0 && !this.alreadyMissAt(down))this.board[this.getRowPos(down.Row)][down.Column-1] = 3;
+            else this.board[this.getRowPos(left.Row)][left.Column-1] = 3;
+            return true;
+        }
+    }
+
+    private verticalShipSunk(lastShot:Position):boolean{
+        let count = 0;
+        let curPos = lastShot;
+        for(let i =1; i<=5; i ++){
+            curPos = curPos.getPositionAbove();
+            if(this.alreadyMissAt(curPos) || this.getRowPos(curPos.Row) ==9)          break;
+            else if(this.alreadyHitAt(curPos))                          count++;
+            else{
+                this.board[this.getRowPos(curPos.Row)][curPos.Column-1] = 3;
+                return true;  
+            }
+        }
+
+        curPos = lastShot;
+        const startCount = count;
+        for(let i=1; i <= 5-startCount; i ++){
+            curPos = curPos.getPositionDown();
+            if(this.alreadyMissAt(curPos) || this.getRowPos(curPos.Row) ==0)          break;
+            else if(this.alreadyHitAt(curPos))                          count++;
+            else{
+                this.board[this.getRowPos(curPos.Row)][curPos.Column-1] = 3;
+                return true;  
+            }
+        }
+
+        return false;
+    }
+
+    private horizontalShipSunk(lastShot:Position):boolean{
+        let count = 0;
+        let curPos = lastShot;
+        for(let i =1; i<=4; i ++){
+            curPos = curPos.getPositionLeft();
+            if(this.alreadyMissAt(curPos) || curPos.Column == 10)          break;
+            else if(this.alreadyHitAt(curPos))                          count++;
+            else{
+                this.board[this.getRowPos(curPos.Row)][curPos.Column-1] = 3;
+                return true;  
+            }
+        }
+
+        const startCount = count;
+
+        curPos = lastShot;
+        for(let i=1; i <= 4-startCount; i ++){
+            curPos = curPos.getPositionRight();
+            if(this.alreadyMissAt(curPos) || curPos.Column== 1)          break;
+            else if(this.alreadyHitAt(curPos))                          count++;
+            else{
+                this.board[this.getRowPos(curPos.Row)][curPos.Column-1] = 3;
+                return true;  
+            }
+        }
+
+        return false;
+    }
+    
+
+    private positionFromCoords(row, col){
+        return new Position(String.fromCharCode(row+65), col+1);
+    }
+
+    public constructor(gameState){
```

Comment:
Two things here.

One - it would be nice to give gameState  a proper type, for the safety of the thing!

Two - it's much easier to follow a class if they all follow the same structure. For us, I'd like to say that roughly, every class should be:

* class variables
* constructor
* public methods
* private methods

...then you always know where to look.

(That's not to say they can't deviate from that - you just need a good reason to make things more surprising to the reader!)